functions/zoom-downloader.py

```diff
@@ -60,96 +68,213 @@ def handler(event, context):
-            "topic": recording_data['topic'],
-            "meeting_start_time": recording_data['start_time'],
-            "recording_start_time": recording_data['recording_files'][0]['recording_start'],
-            "recording_count": recording_data['recording_count'],
-            "webhook_received_time": message_body['received_time'],
-            "correlation_id": message_body['correlation_id']
+            "uuid": self.uuid,
+            "zoom_series_id": self.zoom_series_id,
+            "opencast_series_id": self.opencast_series_id,
+            "host_name": self.data['host'],
+            "topic": self.data['topic'],
+            "created": datetime.strftime(self.created, '%Y-%m-%dT%H:%M:%SZ'),
+            "webhook_received_time": self.data['received_time'],
+            "correlation_id": self.data['correlation_id']
         }
+        return upload_message
 
-        send_to_sqs(upload_message, UPLOAD_QUEUE_NAME)
-    except PermanentDownloadError as e:
-        logger.error(e)
-        send_to_sqs(message_body, DEADLETTER_QUEUE_NAME, error=e)
-        download_message.delete()
-        raise
+    def download(self):
 
-    download_message.delete()
+        if not self.opencast_series_id:
+            logger.info("No opencast series match found")
+            return
+
+        if self.recording_data:
+            logger.info({'recording_data': self.recording_data})
+        else:
+            raise PermanentDownloadError("No recording data found!")
 
+        logger.info(self.recording_data)
 
-def get_host_data(host_id):
+        if not self.chronological_files:
+            raise PermanentDownloadError("No files available to download.")
+        logger.info("downloading {} files".format(len(self.chronological_files)))
 
-    endpoint_url = urljoin(ZOOM_API_BASE_URL, 'users/{}'.format(host_id))
-    host_data = get_api_data(endpoint_url)
-    return {
-        'host_name': "{} {}".format(host_data['first_name'], host_data['last_name']),
-        'host_email': host_data['email']
-    }
+        track_sequence = 1
+        prev_file = None
 
+        for file in self.chronological_files:
+            if next_track_sequence(prev_file, file):
+                track_sequence += 1
+
+            logger.debug("file {} is track sequence {}".format(file, track_sequence))
+            prev_file = file
 
-def get_recording_data(uuid):
-    # Must use string concatenation rather than urljoin because uuids may contain
-    # url unsafe characters like forward slash
-    endpoint_url = ZOOM_API_BASE_URL + 'meetings/{}/recordings'.format(uuid)
-    return get_api_data(endpoint_url, validate_callback=verify_recording_status)
+            stream_file_to_s3(file, self.uuid, track_sequence)
+
+        return self.upload_message
+
+
+def meeting_metadata(uuid):
+    """
+    Get metadata about meeting. Metadata includes zoom meeting id,
+    host_name, host_email, topic, and start_time.
+    """
+    for meeting_type in ['past', 'pastOne', 'live']:
```

Comment:
It would be good if, somewhere in this chain of functions (get_api_data, api_request) each request attempt was logged at at least debug level.

Reply:
or maybe somewhere in the requests/urllib modules it does? if so, nevermind.

Reply:
Ran it to be sure, yes urllib logs the requests.